### PR TITLE
#384-2 lao templates update

### PR DESCRIPTION
**Pull request by andreievg**
Added some changes to get ready for demo server:

### Changes

* Tried to minimise and simplify permissions names and user set for lao data file
* Added further user outcomes
* Added further org outcomes
* Changed default values for object response to be an empty object
* Made populated fields editable (for licence)
* Fixed professional details in licence

### CODE REVIEW

File: database/insertData/laos/02_template_orgRegistration.js [hunk 1]
```diff
@@ -657,7 +656,7 @@ exports.queries = [
                       title: "CV upload"
                       elementTypePluginCode: "fileUpload"
                       category: QUESTION
-                      isRequired: false # TO-DO make required for production
+                      isRequired: true # TO-DO make required for production
```
Comment (nmadruga) [inline database/insertData/laos/02_template_orgRegistration.js:659]: Can remove todo

Comment (nmadruga) [reply database/insertData/laos/02_template_orgRegistration.js:659]: In all next one too : )

File: database/insertData/laos/03_template_license_MMC_MMD.js [hunk 2]
```diff
@@ -436,7 +425,6 @@ exports.queries = [
                         title: "Education List"
                         elementTypePluginCode: "listBuilder"
                         category: QUESTION
-                        isEditable: false
```
Comment (nmadruga) [inline database/insertData/laos/03_template_license_MMC_MMD.js:439]: For all of the above. Isn't it misleading to allow the user to change this, but that doesn't update the record in the system? And if it does, I think the documents that prove need to be re-uploaded (this is regarding "Organisation category" basically)

File: database/insertData/laos/03_template_license_MMC_MMD.js [hunk 3]
```diff
@@ -510,56 +498,169 @@ exports.queries = [
...
+                                {
+                                  operator: "objectProperties"
+                                  children: ["responses.thisResponse"]
+                                }
+                                {
+                                  value: "^[A-Za-z0-9.]+@[A-Za-z0-9]+\\\\.[A-Za-z0-9.]+$"
+                                }
+                              ]
+                            }
+                            validationMessage: "Not a valid email address"
```
Comment (nmadruga) [inline database/insertData/laos/03_template_license_MMC_MMD.js:627]: Same thing about re-uploading documents on this one: 
- **LMMD02** – CV of applicant with recent (max 1 year) photo
- **LMMD06** – Certificate of education level
- **LMMD08** – Letter from previous employers documenting at least 3 years professional experience (for private sector must be certified by provincial or capital level)
- **LMMD09** – Documentation showing resignation from previous employment (for private sector must be certified by provincial or capital level)
- **LMMD09** – Documentation showing resignation from previous employment (for private sector must be certified by provincial or capital level)

File: database/insertData/laos/03_template_license_MMC_MMD.js [hunk 4]
```diff
@@ -1072,7 +1160,6 @@ exports.queries = [
                         elementTypePluginCode: "shortText"
                         category: QUESTION
                         parameters: { label: "Owner email" }
-                        isEditable: false
                         defaultValue: {
                           operator: "buildObject",
                           properties: [
```
Comment (nmadruga) [inline database/insertData/laos/03_template_license_MMC_MMD.js:1165]: Same thing about re-uploading documents (if we allow changing this fields from company/ownership):

### Owner
- **LMMD03** – Medical Certificate (max 3 months)
- **LMMD04** – Certificate of current residence with photo (max 3 months)
- **LMMD05** – Recent (max 1 year) photo
- **LMMD07** – Criminal record "number 3"

### Company
- **LMMD10** – Map of location of company
- **LMMD11** – Layout of facilities
- **LMMD13** – Proof of ownership of facilities or contract of rent
- **LMMD14** – Copy of business licence issued by Ministry of Commerce

**Review by nmadruga — CHANGES_REQUESTED** [review]
Great. Thanks for the final touches!

I realised I hadn't commit the outcomes for licenses yesterday before going home. Sorry about that. Now it's already on `master` so it should be easier for us to test other PR on top of existing templates.

My only concern with the changes here is that allowing the user to edit pre-filled fields would be misleading regarding - this fields aren't updated in the user/organisation table and if they are - users need to re-upload documents since they are making changes to what was reviewed/approved during the company registration process.

**Comment by andreievg** [comment]
> My only concern with the changes here is that allowing the user to edit pre-filled fields would be misleading regarding - this fields aren't updated in the user/organisation table and if they are - users need to re-upload documents since they are making changes to what was reviewed/approved during the company registration process.

Would this be ok (https://github.com/openmsupply/application-manager-web-app/pull/779#issuecomment-858516751, and subsequent comments)

**Comment by nmadruga** [comment]
Sure. That's all good. Have this been implemented here already?

**Review by nmadruga — COMMENTED** [review]
Not sure where is the problem... But I'm now getting this problem when trying to run the `yarn database_init laos`:
Screen Shot 2021-06-14 at 12 24 21 PM

Will be waiting on the fix to test this again.

File: database/insertData/laos/02_template_orgRegistration.js [hunk 5]
```diff
@@ -490,14 +490,13 @@ exports.queries = [
                       }
                     }
                     {
-                      code: "profID"
+                      code: "profId"
                       index: 20
-                      title: "Professional ID"
+                      title: "Pofessional Id"
```
Comment (CarlosNZ) [inline database/insertData/laos/02_template_orgRegistration.js:495]: Spelling -- revert?

File: database/insertData/laos/02_template_orgRegistration.js [hunk 6]
```diff
@@ -182,7 +182,7 @@ exports.queries = [
                         maxLength: 150
                       }
                     }
-                    {
+                    { 1
```
Comment (CarlosNZ) [inline database/insertData/laos/02_template_orgRegistration.js:185]: There's the glitch that is causing the error!

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Just a couple of tiny comments

**Comment by CarlosNZ** [comment]
> Not sure where is the problem... But I'm now getting this problem when trying to run the `yarn database_init laos`:

There's the problem, if you want to fix it for testing: https://github.com/openmsupply/application-manager-server/pull/403#discussion_r651312752

**Comment by nmadruga** [comment]
Great catch @CarlosNZ. I think I'll be fixing the spelling mistake and what you pointed out and will merge this one, since I want to merge my next PR in, and so we try to keep the list of PRs shorter. It gets to a point that is very hard to work like that.

Let me know if any problems to merge in this one @andreievg.

**Comment by CarlosNZ** [comment]
> Great catch @CarlosNZ. I think I'll be fixing the spelling mistake and what you pointed out and will merge this one, since I want to merge my next PR in, and so we try to keep the list of PRs shorter. It gets to a point that is very hard to work like that.
> 
> Let me know if any problems to merge in this one @andreievg.

Yeah sounds good. Merge 👍🏼

**Pull request merged**